src/test/kotlin/reminder/application/ReminderSearcherTest.kt

```diff
@@ -0,0 +1,60 @@
+package reminder.application
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import reminder.domain.Reminder
+import reminder.domain.ReminderIdentifierGenerator
+import reminder.domain.ReminderRepository
+import reminder.mothers.ReminderMother
+import java.util.*
+import kotlin.test.assertEquals
+
+class ReminderSearcherTest {
+    private lateinit var reminderSaver: ReminderSaver
+    private lateinit var generator: ReminderIdentifierGenerator
+    private lateinit var reminderSearcher: ReminderSearcher
+    private lateinit var repository: ReminderRepository
+
+    @BeforeEach
+    fun setUp() {
+        repository = Mockito.mock(ReminderRepository::class.java)
+        generator = Mockito.mock(ReminderIdentifierGenerator::class.java)
+        reminderSearcher = ReminderSearcher(repository)
+        reminderSaver = ReminderSaver(repository, generator)
+    }
+
+    @Test
+    fun `Nothing is returned when valid identifier doesn't exist in the repository`() {
+        val reminder = ReminderMother.getValidReminderWithoutDescription()
+        `Assert that the reminder was not found in the repository given the identifier from`(reminder)
```

Comment:
Here you can call only the ReminderIdentifier mother (since the identifier is the only thing that you are using).

So, the code would be something like:
1) get identifier
2) say mockito that the call should return optional.none
3) assert that mockito was called with the identifier
4) assert that the result is optional.none